### PR TITLE
Update __init__ return typing to be consistent

### PR DESCRIPTION
**Pull request by Harshal6927**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

### CODE REVIEW

**Comment by Goldziher** [comment]
Ok, if we want to go this path we should actually use the `NoReturn` type in all init methods.

**Comment by peterschutt** [comment]
> Ok, if we want to go this path we should actually use the `NoReturn` type in all init methods.

Mypy doesn't let you do that:

```
src/worker/parse.py:14: error: The return type of "__init__" must be None  [misc]
Found 1 error in 1 file (checked 92 source files)
```

Seems [`NoReturn`](https://docs.python.org/3/library/typing.html#typing.NoReturn) is for functions that can never return, e.g., always raise:

> Special type indicating that a function never returns.

Some refs:
- https://github.com/python/mypy/issues/10342#issuecomment-822960700
- https://github.com/microsoft/pyright/issues/3188#issuecomment-1066015281

**Comment by Goldziher** [comment]
Aight

**Comment by Harshal6927** [comment]
I added return type because initializer of MakoTemplateEngine class have None as an return type. It is for the constancy.

**Comment by peterschutt** [comment]
@Harshal6927 why don't we broaden this PR out and annotate the return for all `__init__()` methods that need it, at least that way we are consistent everywhere:

image

**Comment by Harshal6927** [comment]
> @Harshal6927 why don't we broaden this PR out and annotate the return for all `__init__()` methods that need it, at least that way we are consistent everywhere:
> 
> image

Should I do it?

**Comment by peterschutt** [comment]
> Should I do it?

I think so, best to be consistent.

And [pep recommended](https://peps.python.org/pep-0484/#the-meaning-of-annotations):

> (Note that the return type of `__init__` ought to be annotated with -> None. The reason for this is subtle. If `__init__` assumed a return annotation of -> None, would that mean that an argument-less, un-annotated `__init__` method should still be type-checked? Rather than leaving this ambiguous or introducing an exception to the exception, we simply say that `__init__` ought to have a return annotation; the default behavior is thus the same as for other methods.)

**Comment by Harshal6927** [comment]
@peterschutt Ok I'll update my PR.

**Review by jonadaly — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -45,7 +45,6 @@ jobs:
             v1-venv-${{ runner.os }}-${{ matrix.python-version }}
             v1-venv-${{ runner.os }}
       - name: Install dependencies
-        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
```
Comment (Goldziher) [inline .github/workflows/ci.yaml:48]: so this change should be reverted, once done this can be merged.

**Comment by Goldziher** [comment]
@all-contributors add @Harshal6927 code

**Pull request merged**